bee-api/bee-rest-api/src/endpoints/mod.rs

```diff
@@ -71,47 +83,32 @@ where
     }
 
     async fn start(node: &mut N, config: Self::Config) -> Result<Self, Self::Error> {
-        let rest_api_config = config.0;
-        let protocol_config = config.1;
-        let network_id = config.2;
-        let bech32_hrp = config.3;
-
-        let consensus_worker = node.worker::<ConsensusWorker>().unwrap().tx.clone();
-        let tangle = node.resource::<Tangle<N::Backend>>();
-        let storage = node.storage();
-        let message_submitter = node.worker::<MessageSubmitterWorker>().unwrap().tx.clone();
-        let message_requester = node.worker::<MessageRequesterWorker>().unwrap().clone();
-        let requested_messages = node.resource::<RequestedMessages>();
-        let peer_manager = node.resource::<PeerManager>();
-        let network_controller = node.resource::<NetworkCommandSender>();
-        let node_info = node.info();
-        let bus = node.bus();
+        let args = Arc::new(ApiArgs {
+            node_id: config.0,
+            node_key_pair: config.1,
+            rest_api_config: config.2,
+            protocol_config: config.3,
+            network_id: config.4,
+            bech32_hrp: config.5,
+            storage: node.storage(),
+            bus: node.bus(),
+            node_info: node.info(),
+            tangle: node.resource::<Tangle<N::Backend>>(),
+            peer_manager: node.resource::<PeerManager>(),
+            requested_messages: node.resource::<RequestedMessages>(),
+            network_command_sender: node.resource::<NetworkCommandSender>(),
+            message_submitter: node.worker::<MessageSubmitterWorker>().unwrap().tx.clone(),
+            message_requester: node.worker::<MessageRequesterWorker>().unwrap().clone(),
+            consensus_worker: node.worker::<ConsensusWorker>().unwrap().tx.clone(),
+        });
```

Comment:
Do you think we might loose performance by putting all of these into a single `Arc`?

Reply:
Seems like putting "API arguments" into an Arc and passing them to the warp endpoits is the general recommandation.
Instead of cloning only the specific arguments we now clone the Arc in addition.
This 1 reference extra is imo worth it.